### DISCUSSION TITLE
Ngrok share configs per directory

### DISCUSSION THREAD

**Discussion by JacobBennett** [laravel/valet#1118] [Ideas]
https://github.com/laravel/valet/blob/74e203e37b9549b364feda1920e91e3156f38ad9/valet#L32

Would be nice to be able to call `valet share` from any directory, and have valet check to see if there was a `.valet` file in that directory that would store some configs you could use to handle `valet share`.

If using a paid version of ngrok, you can set custom subdomains using a command similar to

```
valet share -suddomain=mysubdomain -region=us
```

if you could store the `subdomain` and the `region` in a `.valet` file per project and check for that when doing a valet share, would be able to drop the extra pieces after `valet share` and have it handled automatically.

sudo code:
```
if [ -f $PWD/.valet ]; then
    CUSTOM_SHARE=$(php "$DIR/cli/valet.php" ngrok_share)
fi
```

**Comment by drbyte** [comment]
I like this idea. 
Undecided what the right name for the config file should be. I think it should be more specific than just `.valet`.  `.valet-ngrok` perhaps?

**Comment by harrison-drift** [comment]
I think `.valet` is fine, personally.

**Comment by sjelfull** [comment]
Could we extend this to have a global default as well?

I use one custom domain for all my sites, and it would be great to have this defined somewhere so I don't have to pass `--hostname` and `--region` to `valet share` every time.

**Comment by drbyte** [comment]
@JacobBennett did you ever do something with this?

**Comment by JacobBennett** [comment]
Never did no. Have at it! :)

**Comment by drbyte** [comment]
@sjelfull how are you handling these parameters today?

**Comment by sjelfull** [comment]
I'm just passing it manually. But some day! :D

**Comment by comes** [comment]
Have a look at https://github.com/laravel/valet/pull/881
